### ISSUE TITLE
Live migration failed when primary node has failed disk

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#329]
Originally reported of Google Code with ID 273.

```
When the primary node for DRBD-instance has a disk failure, so drbd-overview has state Diskless/Secondary, then I expect that

# gnt-instance migrate <instance>

will do live migration to the secondary node.

In fact it is not done: ganeti reports that DRBD has unexpected state "Diskless" for primary node and failing the operation.

A brief code overlook shows (if I understood right) that it is because the normal DRBD operation state, as configured by ganeti, not allowing two primaries, and the migration process is done by spliting the DRBD's parts and reassembling them back with "allow-two-primaries" mode for migration purpose.

As a result, there expecting a short period when the instance is attached to the local storage only, which is offline due to failure.

Formal info:

gnt-cluster (ganeti v2.5.1) 2.5.1

Software version: 2.5.1
Internode protocol: 2050000
Configuration format: 2050000
OS api version: 20
Export interface: 0

OS: Debian 6.0.6

<b>What steps will reproduce the problem?</b>
1. 2-node cluster, each node has separate hard drive for the os and the raid for the LVM/xenvg.

2. Create drbd-backed instance and start it.

3. Fail the drive which serve LVM/xenvg on the primary node.
(I have pull several physical disks off the node).
Check via "drbd-overview" that the disk state is now "Diskless/Secondary".

3. Issue "gnt-instance migrate <instance>"

I got an error reflecting that drbd going to unexpected state.

I expect the flawless live migration, which will help to survive the potentially survivable hardware failure without a downtime.

Best Regards, Dmi.
```
Originally added on 2012-10-01 10:42:35 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This is known and expected. Until we support drbd 8.4, which can do live reconfiguration of network state, this cannot be improved.
```
Originally added on 2012-10-01 10:44:22 +0000 UTC.

Changed State: WontFix

**Comment by Ganeti-Issues-Migrator** [comment]
```
Note: there is issue 212 filled for the support of newer DRBD.
```
Originally added on 2012-10-01 10:48:11 +0000 UTC.

Mentioned Issue with GoogleCode ID 212 has been migrated to GitHub with ID #268.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-01 10:49:12 +0000 UTC.

Now blocked by: 212 (now #268);

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
Unfortunately we found 8.4 buggy and couldn't do on the fly reconfiguration either.
As such we have no fix for this issue.
```
Originally added on 2013-12-09 14:13:22 +0000 UTC.

Added Labels: Type-Defect
Changed State: WontFix

**Comment by Ganeti-Issues-Migrator** [comment]
```
Is it a bug which is known upstream? Any details on the problem?
```
Originally added on 2013-12-10 07:22:18 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
+thomas who was working on that issue :)


```
Originally added on 2013-12-10 08:36:27 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Please see issue 449 for further details and pointers to upstream discussions.
```
Originally added on 2013-12-11 08:39:48 +0000 UTC.

Now blocked by: 449 (now #505);

Mentioned Issue with GoogleCode ID 449 has been migrated to GitHub with ID #505.
